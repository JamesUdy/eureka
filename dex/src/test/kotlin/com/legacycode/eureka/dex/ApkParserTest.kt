package com.legacycode.eureka.dex

import org.approvaltests.Approvals
import org.junit.jupiter.api.Test

class ApkParserTest {
  @Test
  fun `it can build an inheritance register from an APK`() {
    // given
    val apkFile = TestApk("wikipedia.apk").file
    val parser = ApkParser(apkFile)
    val register = parser.inheritanceRegister()

    // when
    val graphvizTree = register.tree(
      Ancestor("Landroid/app/Activity;"),
      DotTreeBuilder("Wikipedia Activities"),
    )

    // then
    Approvals.verify(graphvizTree)
  }
}
