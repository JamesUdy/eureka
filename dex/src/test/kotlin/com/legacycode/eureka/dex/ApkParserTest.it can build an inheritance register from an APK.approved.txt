digraph "Wikipedia Activities" {
  "Lcom/google/android/gms/common/api/GoogleApiActivity;" -> "Landroid/app/Activity;"
  "Landroidx/core/app/ComponentActivity;" -> "Landroid/app/Activity;"
  "Landroidx/activity/ComponentActivity;" -> "Landroidx/core/app/ComponentActivity;"
  "Landroidx/fragment/app/FragmentActivity;" -> "Landroidx/activity/ComponentActivity;"
  "Landroidx/appcompat/app/AppCompatActivity;" -> "Landroidx/fragment/app/FragmentActivity;"
  "Lorg/wikipedia/activity/BaseActivity;" -> "Landroidx/appcompat/app/AppCompatActivity;"
  "Lorg/wikipedia/activity/SingleFragmentActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/categories/CategoryActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/createaccount/CreateAccountActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/edit/EditSectionActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/edit/insertmedia/InsertMediaActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/gallery/GalleryActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/language/LangLinksActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/language/LanguagesListActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/login/LoginActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/login/ResetPasswordActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/notifications/NotificationActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/notifications/NotificationFilterActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/page/PageActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/page/edithistory/EditHistoryListActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/page/tabs/TabActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/settings/AboutActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/settings/LicenseActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/suggestededits/SuggestedEditsImageTagEditActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/suggestededits/SuggestedEditsImageTagsOnboardingActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/talk/ArchivedTalkPagesActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/talk/TalkReplyActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/talk/TalkTopicActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/talk/TalkTopicsActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/usercontrib/UserContribFilterActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/usercontrib/UserContribListActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/watchlist/WatchlistFilterActivity;" -> "Lorg/wikipedia/activity/BaseActivity;"
  "Lorg/wikipedia/activity/SingleFragmentActivityTransparent;" -> "Lorg/wikipedia/activity/SingleFragmentActivity;"
  "Lorg/wikipedia/commons/FilePageActivity;" -> "Lorg/wikipedia/activity/SingleFragmentActivity;"
  "Lorg/wikipedia/descriptions/DescriptionEditActivity;" -> "Lorg/wikipedia/activity/SingleFragmentActivity;"
  "Lorg/wikipedia/descriptions/DescriptionEditTutorialActivity;" -> "Lorg/wikipedia/activity/SingleFragmentActivity;"
  "Lorg/wikipedia/diff/ArticleEditDetailsActivity;" -> "Lorg/wikipedia/activity/SingleFragmentActivity;"
  "Lorg/wikipedia/feed/configure/ConfigureActivity;" -> "Lorg/wikipedia/activity/SingleFragmentActivity;"
  "Lorg/wikipedia/feed/news/NewsActivity;" -> "Lorg/wikipedia/activity/SingleFragmentActivity;"
  "Lorg/wikipedia/feed/onthisday/OnThisDayActivity;" -> "Lorg/wikipedia/activity/SingleFragmentActivity;"
  "Lorg/wikipedia/feed/topread/TopReadArticlesActivity;" -> "Lorg/wikipedia/activity/SingleFragmentActivity;"
  "Lorg/wikipedia/main/MainActivity;" -> "Lorg/wikipedia/activity/SingleFragmentActivity;"
  "Lorg/wikipedia/onboarding/InitialOnboardingActivity;" -> "Lorg/wikipedia/activity/SingleFragmentActivity;"
  "Lorg/wikipedia/page/customize/CustomizeToolbarActivity;" -> "Lorg/wikipedia/activity/SingleFragmentActivity;"
  "Lorg/wikipedia/random/RandomActivity;" -> "Lorg/wikipedia/activity/SingleFragmentActivity;"
  "Lorg/wikipedia/readinglist/ReadingListActivity;" -> "Lorg/wikipedia/activity/SingleFragmentActivity;"
  "Lorg/wikipedia/search/SearchActivity;" -> "Lorg/wikipedia/activity/SingleFragmentActivity;"
  "Lorg/wikipedia/settings/DeveloperSettingsActivity;" -> "Lorg/wikipedia/activity/SingleFragmentActivity;"
  "Lorg/wikipedia/settings/NotificationSettingsActivity;" -> "Lorg/wikipedia/activity/SingleFragmentActivity;"
  "Lorg/wikipedia/settings/SettingsActivity;" -> "Lorg/wikipedia/activity/SingleFragmentActivity;"
  "Lorg/wikipedia/settings/languages/WikipediaLanguagesActivity;" -> "Lorg/wikipedia/activity/SingleFragmentActivity;"
  "Lorg/wikipedia/suggestededits/SuggestionsActivity;" -> "Lorg/wikipedia/activity/SingleFragmentActivity;"
  "Lorg/wikipedia/theme/ThemeFittingRoomActivity;" -> "Lorg/wikipedia/activity/SingleFragmentActivity;"
  "Lorg/wikipedia/watchlist/WatchlistActivity;" -> "Lorg/wikipedia/activity/SingleFragmentActivity;"
  "Lorg/wikipedia/descriptions/DescriptionEditSuccessActivity;" -> "Lorg/wikipedia/activity/SingleFragmentActivityTransparent;"
}
